app/src/test/java/com/mstudio/superheroarch/presentation/CharacterDetailViewModelTest.kt

```diff
@@ -83,4 +86,22 @@ class CharacterDetailViewModelTest {
         viewModel.onCharacterReceived(RickAndMortyRepositoryInstruments.givenACharacterEntity().toCharacterData())
         verify(view).showNoInternetConnection()
     }
+
+    @Test
+    fun `given character detail screen, when user marks character as favorite, then the character is shown as favorite`() = runTest {
+        viewModel.onCharacterReceived(RickAndMortyRepositoryInstruments.givenACharacterEntity().toCharacterData())
+        verify(view).showCharacterAsNonFavorite()
+        `when`(setFavCharacterUseCase.setCharacterAsFavorite(true, 1)).thenReturn(Unit)
```

Comment:
from my point of view is better to order the test in a different way, first put the mocks you are using for this test, then the test itself.

If you add 'when' in the middle of the test, makes more difficult to read and understand what's happening.

It should work as the usecase is called when onFavoriteClicked() is called, not in onCharacterReceived, so the mock should be waiting until is needed.